cpp/src/arrow/compute/kernels/hash.h

```diff
@@ -34,29 +34,10 @@ namespace compute {
 
 class FunctionContext;
 
-/// \brief Invoke hash table kernel on input array, returning any output
-/// values. Implementations should be thread-safe
-class ARROW_EXPORT HashKernel : public UnaryKernel {
```

Comment:
I think we _will_ want to publicly expose this at some point for parallelization. But I suppose it is OK to leave it out of the public header for now

Reply:
Agreed, I didn't want to sign-up for a contract to this until a little more thought it given.  We should probably reorganize the code a little bit in general (I don't think everything having to do with hashing belongs in this header and .cc file).  I think most of the other kernel implementations are also private.